modules/honzon-distribution/src/lib.rs

```diff
@@ -0,0 +1,378 @@
+		let params =
+			DistributionDestinationParams::<T>::get(&destination).ok_or(Error::<T>::DistributionParamsNotExist)?;
+		match destination.clone() {
+			DistributionDestination::StableAsset(stable_asset) => {
+				let amount = Self::adjust_for_stable_asset(&destination, stable_asset, params)?;
+				Self::update_distributed_balance(&destination, amount)?;
+
+				Pallet::<T>::deposit_event(Event::<T>::AdjustDestination { destination, amount });
+			}
+		}
+		Ok(())
+	}
+
+	pub fn do_remove_destination(destination: DistributionDestination<T::AccountId>) -> DispatchResult {
+		let mut params =
+			DistributionDestinationParams::<T>::get(&destination).ok_or(Error::<T>::DistributionParamsNotExist)?;
+		// manual set capacity to zero, trigger redeem all stable asset.
+		params.capacity = 0;
+		match destination.clone() {
+			DistributionDestination::StableAsset(stable_asset) => {
+				let amount = Self::adjust_for_stable_asset(&destination, stable_asset, params)?;
+				DistributedBalance::<T>::remove(&destination);
+				DistributionDestinationParams::<T>::remove(&destination);
+
+				Pallet::<T>::deposit_event(Event::<T>::RemoveDistribution { destination, amount });
+			}
+		}
+		Ok(())
+	}
+
+	/// if current value less than target_min, mint aUSD:
+	///     (balances+x)/(total_supply+x)=target,
+	///     x=(target*total_supply-balances)/(1-target)
+	/// if current value large than target_max, burn aUSD:
+	///     (balances+x)/(total_supply+x)=target,
+	///     x=(balances-target*total_supply)/(1-target)
+	///
+	/// return `Amount` that will be add to or subtract from `DistributedBalance`.
+	fn adjust_for_stable_asset(
+		destination: &DistributionDestination<T::AccountId>,
+		stable_asset: DistributionToStableAsset<T::AccountId>,
+		params: DistributionParams,
+	) -> Result<Amount, DispatchError> {
+		let stable_currency = T::GetStableCurrencyId::get();
+		let pool_id = stable_asset.pool_id;
+		let pool_info = T::StableAsset::pool(pool_id).ok_or(Error::<T>::InvalidDestination)?;
+		let account_id = stable_asset.account_id;
+		let asset_length = pool_info.assets.len();
+		let asset_index = stable_asset.stable_token_index as usize;
+		let total_supply = pool_info.total_supply;
+		let ausd_supply = pool_info
+			.balances
+			.get(asset_index)
+			.ok_or(Error::<T>::InvalidDestination)?;
+		let distributed = DistributedBalance::<T>::get(destination).unwrap_or_default();
+		let capacity = params.capacity;
+
+		let current_rate = Ratio::saturating_from_rational(*ausd_supply, total_supply);
+		let one: Ratio = One::one();
+		if current_rate > params.target_max || capacity < distributed {
+			// current rate large than target_max, or capacity is less than distributed, burn aUSD
+			let burn_amount = if capacity < distributed {
+				let remain = distributed.saturating_sub(capacity).min(*ausd_supply);
+				let stable_balance = T::Currency::free_balance(StableAssetPoolToken(pool_id), &account_id);
+				remain.min(stable_balance)
```

Comment:
in some case when rate of aUSD and lp not maintain in 1. the result of `distributed-capacity` may large than lp share, we use lp share to redeem. because current `redeem_single` accept lp token amount. other case redeem works, although not strict precise, but it's acceptable.